SteamKit2/SteamKit2/Steam/Handlers/SteamUnifiedMessages/SteamUnifiedMessages.cs

```diff
@@ -158,101 +72,79 @@ public AsyncJob<ServiceMethodResponse> SendMessage<TRequest>( string name, TRequ
         public void SendNotification<TRequest>( string name, TRequest message )
             where TRequest : IExtensible, new()
         {
-            if ( message == null )
+            if ( message is null )
             {
                 throw new ArgumentNullException( nameof( message ) );
             }
 
             // Notifications do not set source jobid, otherwise Steam server will actively reject this message
             // if the method being used is a "Notification"
-            var eMsg = Client.SteamID == null ? EMsg.ServiceMethodCallFromClientNonAuthed : EMsg.ServiceMethodCallFromClient;
+            var eMsg = Client.SteamID is null ? EMsg.ServiceMethodCallFromClientNonAuthed : EMsg.ServiceMethodCallFromClient;
             var msg = new ClientMsgProtobuf<TRequest>( eMsg );
             msg.Header.Proto.target_job_name = name;
             msg.Body = message;
             Client.Send( msg );
         }
 
-        /// <summary>
-        /// Creates a <see cref="UnifiedService&lt;TService&gt;"/> wrapper for expression-based unified messaging.
-        /// </summary>
-        /// <typeparam name="TService">The type of a service interface.</typeparam>
-        /// <returns>The <see cref="UnifiedService&lt;TService&gt;"/> wrapper.</returns>
-        public UnifiedService<TService> CreateService<TService>()
-        {
-            return new UnifiedService<TService>( this );
-        }
-
-
-        /// <summary>
-        /// Handles a client message. This should not be called directly.
-        /// </summary>
-        /// <param name="packetMsg">The packet message that contains the data.</param>
+        /// <inheritdoc />
         public override void HandleMsg( IPacketMsg packetMsg )
         {
-            switch ( packetMsg.MsgType )
```

Comment:
This EMsg handling needs to stay, and not solely rely on target_job_name.

`ServiceMethodNotification` needs to stay, because Steam server can send clients notifications too.

Reply:
We should avoid trying to parse the job name before verifying the EMsg imo.